support-frontend/assets/pages/[countryGroupId]/checkout/helpers/getProductFields.ts

```diff
@@ -128,5 +141,10 @@ export const getProductFields = ({
 				fulfilmentOptions: NoFulfilmentOptions,
 				productOptions: NoProductOptions,
 			};
+
+		case 'GuardianPatron':
+		case 'OneTimeContribution':
+			logException(unsupportedProductMessage);
+			throw new Error(unsupportedProductMessage);
```

Comment:
We need to do something with these product keys otherwise TS reports that there are unhandled cases in the switch statement which would result in returning `undefined` from this function. I don't think we'd ever actually hit this case. We don't support these products on the generic checkout and as far as I can tell there will be failures before we get this far.

Reply:
Maybe to aid my understanding. Is there an easy way of determining  where this error would bubble up to? Does it end logging to the console if not handled anyway?

Reply:
The `logException` call reports it to Sentry. We're not actively alarming on it, but at least I can keep an eye out after merging. I'm definitely not expecting this code path to be reached.